### PR TITLE
Don't produce a follow-on error message if we try to perform an impl lookup during error recovery.

### PR DESCRIPTION
**Pull request by zygoloid**

### CODE REVIEW

File: toolchain/check/impl_lookup.cpp [hunk 1]
```diff
@@ -8,12 +8,14 @@
...
 #include "toolchain/sem_ir/ids.h"
+#include "toolchain/sem_ir/typed_insts.h"
 
 namespace Carbon::Check {
 
 static auto FindAssociatedImportIRs(Context& context,
                                     SemIR::ConstantId type_const_id,
-                                    SemIR::ConstantId interface_const_id)
+                                    SemIR::ConstantId interface_const_id,
+                                    bool* any_errors)
```
Comment (chandlerc) [inline toolchain/check/impl_lookup.cpp]: Would it make sense to instead track this in the `Check::Context` as I can imagine an almost limitless range of places where we'd like to know "are we doing error recovery"?

I'm imagining a scoped-error-recovery object that sets a flag in the context and unsets it when going out of scope?

Hmm, but that won't help here.

Is there a stack that we could reasonably push and pop to reliably un-set this flag?

Even if not, the bool being in the context seems better than threading it through APIs to me.

Comment (zygoloid) [reply toolchain/check/impl_lookup.cpp]: This isn't tracking a general "are we doing error recovery" state, it's checking specifically whether the impl query we're performing depends on an error. I think that's the better thing to check for here, as it means we'll be able to type-check more non-error-dependent code after an error.

That said, thinking about this more, I think there's not actually any need to do this as part of the traversal here -- errors should have been propagated to the top level of the query already so we can just check there.

Comment (zygoloid) [reply toolchain/check/impl_lookup.cpp]: Simplified on that basis.

Comment (chandlerc) [reply toolchain/check/impl_lookup.cpp]: Ah, much nicer, thanks!

**Review by chandlerc — APPROVED** [review]
LG!

**Pull request merged**